obspy/io/nied/knet/core.py

```diff
@@ -0,0 +1,312 @@
+# -*- coding: utf-8 -*-
+"""
+Reading of the K-NET and KiK-net ASCII format as defined on
+http://www.kyoshin.bosai.go.jp.
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA @UnusedWildImport
+
+from obspy import UTCDateTime, Stream, Trace
+from obspy.core.trace import Stats
+from datetime import datetime
+import re
+import numpy as np
+
+
+class KnetFormatError(Exception):
+    pass
+
+
+class KnetDataError(Exception):
+    pass
+
+
+def _buffer_proxy(filename_or_buf, function, reset_fp=True,
+                  file_mode="rb", *args, **kwargs):
+    """
+    Calls a function with an open file or file-like object as the first
+    argument. If the file originally was a filename, the file will be
+    opened, otherwise it will just be passed to the underlying function.
+
+    :param filename_or_buf: File to pass.
+    :type filename_or_buf: str, open file, or file-like object.
+    :param function: The function to call.
+    :param reset_fp: If True, the file pointer will be set to the initial
+        position after the function has been called.
+    :type reset_fp: bool
+    :param file_mode: Mode to open file in if necessary.
+    """
+    try:
+        position = filename_or_buf.tell()
+        is_buffer = True
+    except AttributeError:
+        is_buffer = False
+
+    if is_buffer is True:
+        ret_val = function(filename_or_buf, *args, **kwargs)
+        if reset_fp:
+            filename_or_buf.seek(position, 0)
+        return ret_val
+    else:
+        with open(filename_or_buf, file_mode) as fh:
+            return function(fh, *args, **kwargs)
+
+
+def _is_knet_ascii(filename_or_buf):
+    """
+    Checks if the file is a valid K-NET/KiK-net ASCII file.
+
+    :param filename_or_buf: File to test.
+    :type filename_or_buf: str or file-like object.
+    """
+    try:
+        return _buffer_proxy(filename_or_buf, __is_knet_ascii, reset_fp=True)
+    # Happens for example when passing the data as a string which would be
+    # interpreted as a filename.
+    except (OSError, UnicodeDecodeError):
+        return False
+
+
+def __is_knet_ascii(buf):
+    """
+    Checks if the file is a valid K-NET/KiK-net ASCII file.
+
+    :param buf: File to read.
+    :type buf: Open file or open file like object.
+    """
+    first_string = buf.read(11).decode()
```

Comment:
No, it defaults to system encoding, which may or may not be utf-8. And is the text really utf-8?